src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -244,10 +244,16 @@ public void stop(Throwable cause) {
          */
         private String cookie;
 
+        /**
+         * Unique cookie set when the task is queued. Used for status lookups.
+         */
+        private final String queueCookie;
```

Comment:
I think this can be null if deserialized or loading an old PlaceHolderTask -- we should be using a synchronized getter that creates the queueCookie if null.